lumicks/pylake/kymotracker/detail/peakfinding.py

```diff
@@ -76,6 +76,12 @@ def peak_estimate(data, half_width, thresh):
     thresh : float
         Threshold for accepting something as a peak.
     """
+    if thresh <= (minimum_value := np.min(data)):
+        raise RuntimeError(
+            f"Threshold ({thresh}) cannot be lower than or equal to the lowest filtered "
+            f"pixel ({minimum_value})"
+        )
```

Comment:
Though this is the correct way of handling it, it leaves the consumer of that API in a bit of a pickle as they now need to perform extra steps to ensure `threshold` is not in conflict with the supplied data. IMO, `threshold` is more of a default anyway (in how I would use it), so I'd also be fine if the algorithm told me that my default wasn't good enough for the supplied and that instead it chose a new default for me (here: making `np.min(data)` the new `threshold`).

Reply:
I'd be open to that, if we had a clear way of choosing a better one that doesn't exclude some use cases.

In this case, `min` is still not an acceptable input and `eps` over `min` also won't be very good. Giving a value somewhere arbitrary in the range seems very dangerous, since now you get bumped to some special value when you go too low.

I'd rather error out early than spend a really long time tracking nonsense.